interfaces/README.md

```diff
@@ -143,3 +143,94 @@ Run these commands from every app's own 'root'-folder.
+
+- Testing class variables and objects, when they are being defined or constructed
+- There are several methods which serve the purpose of defining class wide variables, which we should also test and verify. One of the typical examples of one such method is `ngOnInit`
+
+```ts
+it('ngOnInit: should set up variables', () => {
+  expect(component.isLoggedIn).toBeDefined(); // check for class variables to be defined
+  expect(component.someValye).toBeTruthy();  // check for a variable to be TRUE
+  expect(component.someValye).toBeFalsy();  // check for a variable to be FALSE
+});
+```
+
+The methods written as `toBeTruthy` are called matchers, they help us compare the expected values, their types, whether a method was called, the arguments of the methods and also their existence. There are various methods provided by the testing module. We can find a detailed list of those methods and their usage here: https://jasmine.github.io/api/3.5/matchers.html
+
+A short introduction tutorial, to start off writing test cases can be found at: https://jasmine.github.io/tutorials/your_first_suite
+
+
+#### Testing method callbacks and changes
+-  In order to test for methods to have been called, or been called with certain arguments use `spy` and `toHaveBeenCalled`/ `toHaveBeenCalledWith` matchers.
+
+```ts
+
+//Code
+function  doLogin (params:type) {
+    event.preventDefault()
+}
+
+//Test
+it('some_method: should call another fn', () => {
+  spyOn(event, "preventDefault"); // Monitor the said method
+  component.doLogin(event); // call some_method
+  expect(event.preventDefault).toHaveBeenCalled(); // check for the monitored method to have been called
+});
+```
+
+#### Testing conditional statements
+-  Make separate `it` blocks for different conditions.
+```ts
+it("Test when xyz === 'some-value'", () => {})
+it("Test when xyz !== 'some-value'", () => {})
+```
+
+#### Testing Async Mehthods (i.e. methods which make an API call)
+-  Make a Spy for the specific async call which returns a Promise object. For example a method containing a call routine `this.programsService.changePassword` can be spied using following
+
+        let spy = spyOn(component.programsService, "changePassword").and.returnValue(Promise.resolve(true));
+
+
+-  Based on the changes / executions upon the completion of the async request, we should aim to test the changes and modifications.
+
+        spy.calls.mostRecent().returnValue.then(() => { // block to test what happenes after the async calls.
+            // Here goes expectations and changes
+        });
+
+-  Make sure the `done()` method is used to account for the async calls and fake async stubs/spies.
+
+        it('XYZ', (done) => {
+          // spies and stubs
+
+          spy.calls.mostRecent().returnValue.then(() => {
+              // tests
+              done(); // to complete the tests
+          });
+        });
```

Comment:
````suggestion
```ts
it('XYZ', (done) => {
  // spies and stubs

  spy.calls.mostRecent().returnValue.then(() => {
    // tests
    done(); // to complete the tests
  });
});
```
````